core/src/test/java/io/questdb/EmbeddedApiTest.java

```diff
@@ -158,4 +165,129 @@ public void testReadWrite() throws Exception {
+                    ) {
+                        try (RecordCursorFactory factory = compiler.compile("abc", ctx).getRecordCursorFactory()) {
+                            try (RecordCursor cursor = factory.getCursor(ctx)) {
+                                final Record record = cursor.getRecord();
+                                //noinspection StatementWithEmptyBody
+                                while (cursor.hasNext()) {
+                                    // access 'record' instance for field values
+                                }
+                            }
+                        }
+                    } catch (SqlException | EntryLockedException e) {
+                        errors.incrementAndGet();
+                        e.printStackTrace();
+                    }
+                }
+                latch.countDown();
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private static class Writer extends Thread {
+
+        private final CairoEngine engine;
+        private final AtomicInteger errors;
+        private final CyclicBarrier barrier;
+        private final CountDownLatch latch;
+        private final int iterations;
+
+        private Writer(CairoEngine engine, AtomicInteger errors, CyclicBarrier barrier, CountDownLatch latch, int iterations) {
+            this.engine = engine;
+            this.errors = errors;
+            this.barrier = barrier;
+            this.latch = latch;
+            this.iterations = iterations;
+        }
+
+        @Override
+        public void run() {
+            try {
+                barrier.await();
+                for (int i = 0; i < iterations; i++) {
+                    try (
+                            final SqlExecutionContextImpl ctx = new SqlExecutionContextImpl(engine, 1);
+                            final SqlCompiler compiler = new SqlCompiler(engine)
+                    ) {
+                        compiler.compile("create table if not exists abc (a int, b byte, ts timestamp) timestamp(ts)", ctx);
+                        try (TableWriter writer = engine.getWriter(ctx.getCairoSecurityContext(), "abc", "testing")) {
+                            for (int j = 0; j < 100; j++) {
+                                TableWriter.Row row = writer.newRow(Os.currentTimeMicros());
+                                row.putInt(0, j);
+                                row.putByte(1, (byte) j);
+                                row.putDate(2, System.currentTimeMillis());
+                                row.append();
+                            }
+                            writer.commit();
+                        }
+                    } catch (SqlException e) {
+                        errors.incrementAndGet();
+                        e.printStackTrace();
+                    }
+                }
+                latch.countDown();
+            } catch (Exception e) {
+                e.printStackTrace();
```

Comment:
could we perhaps bump done latch and error count here too? Or may be restructure exception handling ?

Reply:
Sounds reasonable. Did that in d31c145608446cf494d308002c84771b4e1e8f65

Reply:
thank you Andrey 👍